### ISSUE TITLE
Unable to pick up libpython built with pymalloc

### ISSUE THREAD

**Issue by jakirkham** [pyinstaller/pyinstaller#2053]
In [PEP 3149]( https://www.python.org/dev/peps/pep-3149/ ), it appears that subscript letters were included with `libpython`s that are built. For instance, `m` was provided for pymalloc. This seems to be the default build of Python one will get. However, it seems that PyInstaller will only look for `libpython3.5.dylib` when it should also check for `libpython3.5m.dylib` and probably do so first. Here's a traceback demonstrating the failure.

```
pyinstaller statuspage.spec
569 INFO: PyInstaller: 3.2
570 INFO: Python: 3.5.0
576 INFO: Platform: Darwin-13.4.0-x86_64-i386-64bit
582 INFO: UPX is not available.
585 INFO: Extending PYTHONPATH with paths
['/zopt/conda3/conda-bld/work/statuspage-0.3-pre',
 '/zopt/conda3/conda-bld/work/statuspage-0.3-pre']
585 INFO: checking Analysis
585 INFO: Building Analysis because out00-Analysis.toc is non existent
585 INFO: Initializing module dependency graph...
587 INFO: Initializing module graph hooks...
589 INFO: Analyzing base_library.zip ...
6972 INFO: running Analysis out00-Analysis.toc
6992 INFO: Caching module hooks...
6995 INFO: Analyzing statuspage.py
7005 INFO: Loading module hooks...
7005 INFO: Loading module hook "hook-xml.py"...
7406 INFO: Loading module hook "hook-encodings.py"...
7419 INFO: Loading module hook "hook-pydoc.py"...
7438 INFO: Looking for ctypes DLLs
7438 INFO: Analyzing run-time hooks ...
7454 INFO: Looking for dynamic libraries
7822 INFO: Looking for eggs
7822 INFO: Python library not in binary depedencies. Doing additional searching...
Traceback (most recent call last):
  File "/zopt/conda3/envs/_build/bin/pyinstaller", line 6, in <module>
    sys.exit(PyInstaller.__main__.run())
  File "/zopt/conda3/envs/_build/lib/python3.5/site-packages/PyInstaller/__main__.py", line 90, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "/zopt/conda3/envs/_build/lib/python3.5/site-packages/PyInstaller/__main__.py", line 46, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "/zopt/conda3/envs/_build/lib/python3.5/site-packages/PyInstaller/building/build_main.py", line 788, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "/zopt/conda3/envs/_build/lib/python3.5/site-packages/PyInstaller/building/build_main.py", line 734, in build
    exec(text, spec_namespace)
  File "<string>", line 16, in <module>
  File "/zopt/conda3/envs/_build/lib/python3.5/site-packages/PyInstaller/building/build_main.py", line 212, in __init__
    self.__postinit__()
  File "/zopt/conda3/envs/_build/lib/python3.5/site-packages/PyInstaller/building/datastruct.py", line 178, in __postinit__
    self.assemble()
  File "/zopt/conda3/envs/_build/lib/python3.5/site-packages/PyInstaller/building/build_main.py", line 543, in assemble
    self._check_python_library(self.binaries)
  File "/zopt/conda3/envs/_build/lib/python3.5/site-packages/PyInstaller/building/build_main.py", line 626, in _check_python_library
    raise IOError(msg)
OSError: Python library not found: libpython3.5.dylib, .Python, Python
This would mean your Python installation doesn't come with proper library files.
This usually happens by missing development package, or unsuitable build parameters of Python installation.

* On Debian/Ubuntu, you would need to install Python development packages
  * apt-get install python3-dev
  * apt-get install python-dev
* If you're building Python by yourself, please rebuild your Python with `--enable-shared` (or, `--enable-framework` on Darwin)
```

Using versions Python 3.5 and PyInstaller 3.2.

**Comment by jakirkham** [comment]
I've also tried a few tricks like symlinking, setting fallback paths, etc. No luck thus far. I'm hardly a PyInstaller expert. So, if someone knows something simple I'm missing, I would really appreciate some advice.

**Comment by htgoebel** [comment]
This was fixed in 501534f76b4a4. Please try the current development head.

**Comment by jakirkham** [comment]
That does seem to fix it. Thanks @htgoebel.
